src/arcade/core/util/Graph.java

```diff
@@ -377,108 +545,124 @@ public String toString() {
             }
             s += "\n";
         }
-        
+
         return s;
     }
-    
+
     /**
      * Nested class representing a graph node.
      * <p>
      * The node tracks its corresponding position in the lattice.
      */
-    public abstract static class Node implements Comparable<Node> {
+    public static class Node implements Comparable<Node> {
         /** Coordinate in x direction. */
         protected int x;
-        
+
         /** Coordinate in y direction. */
         protected int y;
-        
+
         /** Coordinate in z direction. */
         protected int z;
-        
+
         /**
          * Creates a {@code Node} at the given coordinates.
          *
-         * @param x  the x coordinate
-         * @param y  the y coordinate
-         * @param z  the z coordinate
+         * @param x
+         *            the x coordinate
+         * @param y
+         *            the y coordinate
+         * @param z
+         *            the z coordinate
          */
         public Node(int x, int y, int z) {
```

Comment:
You might be able to get away with using [Mason's Int3D](https://cs.gmu.edu/~eclab/projects/mason/classdocs.22/sim/util/Int3D.html) as your Nodes instead of having a bespoke class